spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,27 +1,140 @@
+  let(:ems_vmware) { FactoryGirl.create(:ems_vmware, :name => 'test_vmware') }
+  let(:cluster) { FactoryGirl.create(:cluster, :name => 'test_cluster') }
+  let(:user) { FactoryGirl.create(:user) }
   before { stub_user(:features => :all) }
 
-  describe '#tree_select' do
+  context 'render_views' do
     render_views
 
-    context 'switch' do
-      let(:ems) { FactoryGirl.create(:ems_vmware) }
-      let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
-      let(:host) { FactoryGirl.create(:host_vmware, :ems_id => ems.id, :ems_cluster => cluster) }
-      let(:switch) do
-        sw = FactoryGirl.create(:switch_vmware, :ems_id => ems.id)
-        FactoryGirl.create(:host_switch, :host => host, :switch => sw)
-        sw
+    context '#explorer' do
+      before do
+        session[:settings] = {:views => {}, :perpage => {:list => 5}}
+        EvmSpecHelper.create_guid_miq_server_zone
+      end
+
+      it 'can render the explorer' do
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+        get :explorer
+        expect(response.status).to eq(200)
+        expect(response.body).to_not be_empty
+      end
+
+      it 'shows a switch in the list' do
+        switch
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+
+        get :explorer
+        expect(response.body).to include("modelName: 'Switch'")
+        expect(response.body).to include("activeTree: 'infra_networking_tree'")
+        expect(response.body).to include("gtlType: 'list'")
+        expect(response.body).to include("isExplorer: 'true' === 'true' ? true : false")
+        expect(response.body).to include("showUrl: '/infra_networking/x_show/'")
+      end
+    end
+
+    context "#tree_select" do
+      before do
+        switch
+      end
+
+      [
+        ['All Distributed Switches', 'infra_networking_tree'],
+      ].each do |elements, tree|
+        it "renders list of #{elements} for #{tree} root node" do
+          session[:settings] = {}
+          seed_session_trees('infra_networking', tree.to_sym)
+
+          post :tree_select, :params => { :id => 'root', :format => :js }
+          expect(response.status).to eq(200)
+        end
+      end
+
+      it "renders textual summary for an infrastructure switch" do
+        user
+        allow(switch).to receive(:tagged_with).with(:cat => user.userid).and_return("my tags")
+        classification = FactoryGirl.create(:classification, :name => "department", :description => "Department")
+        tag1 = FactoryGirl.create(:classification_tag,
+                                  :name   => "tag1",
+                                  :parent => classification)
+        tag2 = FactoryGirl.create(:classification_tag,
+                                  :name   => "tag2",
+                                  :parent => classification)
```

Comment:
You generate the same things as below, just in an example. Share as much as you can if possible.